src/include/btree.h

```diff
@@ -280,15 +280,16 @@ struct __wt_btree {
 #define WT_BTREE_CLOSED 0x0004000u          /* Handle closed */
 #define WT_BTREE_GARBAGE_COLLECT 0x0008000u /* Content becomes obsolete automatically */
 #define WT_BTREE_IGNORE_CACHE 0x0010000u    /* Cache-resident object */
-#define WT_BTREE_IN_MEMORY 0x0020000u       /* Cache-resident object */
+#define WT_BTREE_IN_MEMORY 0x0020000u       /* In-memory object */
 #define WT_BTREE_LOGGED 0x0040000u          /* Commit-level durability without timestamps */
 #define WT_BTREE_NO_CHECKPOINT 0x0080000u   /* Disable checkpoints */
-#define WT_BTREE_OBSOLETE_PAGES 0x0100000u  /* Handle has obsolete pages */
-#define WT_BTREE_READONLY 0x0200000u        /* Handle is readonly */
-#define WT_BTREE_SALVAGE 0x0400000u         /* Handle is for salvage */
-#define WT_BTREE_SKIP_CKPT 0x0800000u       /* Handle skipped checkpoint */
-#define WT_BTREE_UPGRADE 0x1000000u         /* Handle is for upgrade */
-#define WT_BTREE_VERIFY 0x2000000u          /* Handle is for verify */
+#define WT_BTREE_NO_EVICT 0x0100000u        /* Cache-resident object */
```

Comment:
Reading through these, I can see myself forgetting the difference between `IN_MEMORY` and `NO_EVICT`, especially because in-memory would seem to imply no eviction. What about something that encodes the meaning a little better? `WT_BTREE_NO_CKPT` might be OK but it's already used. Perhaps `WT_BTREE_GARBAGE_COLLECTED`. (Alternatively, a comment above that contains more or less the same information as this PR's cover letter.)

Reply:
IN_MEMORY is different to no eviction. In memory database also do eviction but it doesn't remove it from memory. The same here.

Reply:
Right, but that's not obvious just from reading the definitions here. So either a comment or better names would be useful to me.

Reply:
Closing this out - the latest commits help with this.